src/main/java/com/starrocks/connector/spark/sql/schema/InferSchema.java

```diff
@@ -97,14 +97,15 @@ static StructField inferStructField(StarRocksField field) {
 
         return new StructField(field.getName(), dataType, true, Metadata.empty());
     }
+
     static DataType inferDataType(StarRocksField field) {
         String type = field.getType().toLowerCase(Locale.ROOT);
         switch (type) {
             case "tinyint":
                 // mysql does not have boolean type, and starrocks `information_schema`.`COLUMNS` will return
                 // a "tinyint" data type for both StarRocks BOOLEAN and TINYINT type, We distinguish them by
                 // column size, and the size of BOOLEAN is null
-                return field.getSize() == null ? DataTypes.BooleanType : DataTypes.ByteType;
+                return (field.getSize() == null || field.getSize() == 0) ? DataTypes.BooleanType : DataTypes.ByteType;
```

Comment:
When will the `field.getSize()` be 0 for boolean? Isn't always null?

Reply:
yes. fixed